open-sdg/sdg-build
# 1.0.0-rc1 - default to using the object-oriented API

This PR started out as an attempt at switching to the object-oriented API by default, but has grown to include a more thorough preparation for version 1.0.0. Here is a summary of what is included in this PR:

1. When the legacy Open SDG helper functions are used, now the object-oriented API will be used. These legacy helper functions are `build_data`, `check_all_csv`, and `check_all_meta`. They should continue to function as before (with one exception - see breaking changes below) but will be using the object-oriented approach behind the scenes.
2. This can be seen as a big step towards a 1.0.0 release, so this also bumps the version of the library to 1.0.0-rc1.
3. Open SDG configuration files: This feature adds the capability of managing the configuration of an Open SDG implementation via a config file. This feature is intended to make it easier for countries using the out-of-the-box CSV/YAML approach to tweak their configuration, without requiring them to understand Python code. For an example of the options, see `docs/examples/open_sdg_config.yml`. For an example of using the config file, see `docs/examples/open_sdg_simple.py`.
4. This PR also includes a refresh of the testing metadata - pulling in the latest metadata from open-sdg-data-starter.
5. Some classes, scripts, and tests that are deprecated by the object-oriented approach have been removed, for clarity.
6. Update the default translation repo from sdg-translations to multiple: translations-un-sdg and translations-open-sdg.

## How to test

The best way to test would be to simply point to this branch (`brockfanning/sdg-build-1@default-to-new-api`) in an Open SDG data repository's requirements.txt file. If all goes well, it should work as a "drop in replacement" (but see the breaking changes below).

## Breaking changes

1. The `build_data` helper function no longer supports the `git` or `git_data_dir` parameters. The new behavior is to run the function as if `git` were True. For most implementations this should not present a problem, but please open an issue in GitHub if you encounter any unintended consequences.
2. Because this switches to a new (slightly more strict) validation system, you may find that your metadata fails validation after updating. However the failure messages should include details on which indicators failed and why.
